docs/articles/getting-started/recovery.md

```diff
@@ -0,0 +1,66 @@
+One of the powers of Bytewax is the ability to develop dataflows that
+accumulate state as they process data. By default, this state is just
+stored in memory and will be lost if the dataflow is restarted or
+there is a fault. Bytewax provides some tools and techniques to allow
+you to **recover** your state and resume the dataflow after a fault.
+
+## Prerequisites
+
+Not all dataflows are possible to be recovered. There are a few
+requirements:
+
+1. _Stateful Operators_ - Only the operators
+   [reduce](./operators#reduce) and [stateful
+   map](./operators#stateful_map) aggregate state _between_ epochs. If
+   your dataflow is simple enough to not be using these,
+
+2. _Regular Epochs_ - The [epoch](/getting-started/epochs) is the minimum unit of
+   recovery. Design your dataflow so that epochs are regularly
+   occuring because it is not possible to coherently resume processing
+   mid-epoch.
+
+3. _Replayable Input_ - Your data source needs to support re-playing
+   input from a specific epoch in the past, we'll call the **recovery
+   epoch**.
+   
+4. _At-least-once Output_ - Bytewax only provides at-least-once
+   guarantees when sending data into a downstream system when
+   performing a recovery. You should design your architecture to
+   support this via some sort of idempotency.
+   
+## Enabling Recovery
+
+Your dataflow needs a few features to enable recovery on Bytewax:
+
+1. Design your [input builder](/getting-started/execution#builders) to be able to
+   resume and replay all data from a specific epoch onwards. You are
+   given the freedom to implement this however you'd like.
+
+2. Create a **recovery config** for your workers to know how to
+   connect to the **recovery store**, the database or filesystem that
+   will store recovery data. See our API docs for `bytewax.recovery`
+   on our supported options.
+   
+3. Pass the recovery config as the `recovery_config` argument to
+   whatever execution [entry point](/getting-started/execution)
+   (e.g. `cluster_main()`) you are using.
+   
+## Performing Recovery
+
+If a dataflow has failed, follow this high-level game plan to recover
+and have it resume processing from where it failed:
+
+1. Terminate the dataflow processes.
+
+2. Observe the previous dataflow output and note the final epoch
```

Comment:
We will throw an error if you do not include a `capture()` step when building your dataflow. But, yes, the user could ignore it and never log it our output it and that would make recovery tough. I'll add a comment to log it somewhere. Figuring out the recovery epoch is important enough that we are going to build some machinery to calculate it automatically (also it allows us to garbage collect old recovery state) so I was thinking this would be somewhat of a temporary manual step.